### PR TITLE
fix: catch and toss errors from illegal header names in HARs

### PR DESCRIPTION
**Pull request by erunion**
If a HAR has a header with an illegal name, like `X-API-KEY (Header)`, `Headers.append()` will throw an error. Instead of letting `fetch-har` fail on that we're going to instead catch and toss them.

6e8b0704-c419-4cb4-8305-2718e74a5f49

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -25,6 +26,19 @@ describe('#fetch', () => {
     mock.done();
   });
 
+  it('should catch and toss invalid headers present in a HAR', async () => {
+    const mock = nock('https://httpbin.org')
+      .post('/post')
+      .reply(function () {
+        expect(this.req.headers['x-api-key']).toStrictEqual(['asdf1234']);
+        expect(this.req.headers['x-api-key (invalid)']).toBeUndefined();
```
Comment (domharrington) [inline __tests__/index.test.js:34]: TIL this is an invalid header name: `x-api-key (invalid)`

Comment (domharrington) [reply __tests__/index.test.js:34]: Interesting that curl doesnt appear to block it, but this looks to be an error from httpbin?

```
$ curl -H "x-api-key (invalid): test" https://httpbin.org/headers
<html>
  <head>
    <title>Bad Request</title>
  </head>
  <body>
    <h1><p>Bad Request</p></h1>
    Invalid HTTP header name: &#x27;X-API-KEY (INVALID)&#x27;
  </body>
</html>
```

File: index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 const parseDataUrl = require('parse-data-url');
 
 function constructRequest(har, userAgent = false) {
-  if (!har) throw new Error('Missing HAR file');
+  if (!har) throw new Error('Missing HAR definition');
```
Comment (domharrington) [inline index.js:5]: Does this make it a major version bump?

**Review by domharrington — APPROVED** [review]

File: __tests__/index.test.js (hunk 1 above)
Comment (erunion) [reply __tests__/index.test.js:34]: Postman fails on it too before you can make a request.

Screen Shot 2021-07-29 at 4 49 48 PM

File: index.js (hunk 2 above)
Comment (erunion) [reply index.js:5]: Yeah this'll be a major bump because of the header fix and also some other dev deps I updated.

Comment (erunion) [reply index.js:5]: And I'm also dropping support for Node 10*

File: __tests__/index.test.js (hunk 1 above)
Comment (erunion) [reply __tests__/index.test.js:34]: Can also see this locally:

Screen Shot 2021-07-29 at 4 52 58 PM

cURL must not be doing header name validation? 🤷

**Pull request merged**